batch/batch/server/server.py

```diff
@@ -813,87 +883,29 @@ def update_job_with_pod(job, pod):
 
     log.info('k8s state refresh complete')
 
-    return web.Response(status=204)
-
-
-@routes.get('/recent')
-@aiohttp_jinja2.template('recent.html')
-async def recent(request):  # pylint: disable=W0613
-    recent_events = get_recent_events()
-    return {'recent': list(reversed(recent_events))}
-
-
-def run_forever(target, *args, **kwargs):
-    expected_retry_interval_ms = 15 * 1000
-
-    while True:
-        start = time.time()
-        run_once(target, *args, **kwargs)
-        end = time.time()
-
-        run_time_ms = int((end - start) * 1000 + 0.5)
-
-        sleep_duration_ms = random.randrange(expected_retry_interval_ms * 2) - run_time_ms
-        if sleep_duration_ms > 0:
-            log.debug(f'run_forever: {target.__name__}: sleep {sleep_duration_ms}ms')
-            time.sleep(sleep_duration_ms / 1000.0)
-
-
-def aiohttp_event_loop(port):
-    app.add_routes(routes)
-    web.run_app(app, host='0.0.0.0', port=port)
-
-
-def kube_event_loop(port):
-    # May not be thread-safe; opens http connection, so use local version
-    v1_ = kube.client.CoreV1Api()
-
-    watch = kube.watch.Watch()
-    stream = watch.stream(
-        v1_.list_namespaced_pod,
-        POD_NAMESPACE,
-        label_selector=f'app=batch-job,hail.is/batch-instance={instance_id}')
-    for event in stream:
-        pod = event['object']
-        name = pod.metadata.name
-        requests.post(f'http://127.0.0.1:{port}/pod_changed', json={'pod_name': name}, timeout=120)
-
 
-def polling_event_loop(port):
-    time.sleep(1)
+async def polling_event_loop():
+    await asyncio.sleep(1)
     while True:
         try:
-            response = requests.post(f'http://127.0.0.1:{port}/refresh_k8s_state', timeout=120)
-            response.raise_for_status()
-        except requests.HTTPError as exc:
-            log.error(f'Could not poll due to exception: {exc}, text: {exc.response.text}')
+            await refresh_k8s_state()
         except Exception as exc:  # pylint: disable=W0703
+            traceback.print_exc()
```

Comment:
why not log exception here? internally logging should use `traceback.print_exception()` - using the stack trace provided by `sys.exc_info()` - which `print_exc()` is shorthand for

https://docs.python.org/3/library/logging.html

Reply:
no reason, added this to debug the race condition. changed to exception